docs/autofixture/extending-autofixture.md

```diff
@@ -0,0 +1,157 @@
+        {		
+            return new SampleValueObject
+            {
+                StringValue = context.Resolve(typeof(string)) as string,
+                IntValue = 42
+            };
+        }
+        
+        return new NoSpecimen();
+    }
+}
+```
+
+Specimen builder are added to a given fixture via its Customizations collection.
+
+```csharp
+[Test]
+public void Fixture_uses_specimen_builder_to_create_value()
+{
+    // ARRANGE
+    var fixture = new Fixture();
+    fixture.Customizations.Add(new SampleValueObjectSpecimenBuilder());
+
+    // ACT
+    var obj = fixture.Create<SampleValueObject>();
+
+    // ASSERT
+    Assert.That(obj.IntValue, Is.EqualTo(42));
+}
+```
+
+## Behaviors
+
+Together with specimen builders, behaviors are part of the graph used to represent the chain of responsibility used to create values.
+
+Unlike the specimen builders, that can be seen as the leaves of the graph and therefore can only respond to a request, behaviors have access to both the request and the response and can act or modify them.
+
+For example, the `TracingBehavior` can be used to track the chain of calls used to serve a certain request.
+
+```csharp
+var fixture = new Fixture();
+fixture.Behaviors.Add(new TracingBehavior());
+var value = fixture.Create<SampleValueObject>();
+```
+
+When executing the snippet above, the following will be printed on the standard out
+
+```
+  Requested: AutoFixture.Kernel.SeededRequest
+    Requested: SampleValueObject
+      Requested: System.String StringValue
+        Requested: AutoFixture.Kernel.SeededRequest
+          Requested: System.String
+          Created: d2f151c4-afa6-414c-8168-07b4de1ad5ae
+        Created: StringValued2f151c4-afa6-414c-8168-07b4de1ad5ae
+      Created: StringValued2f151c4-afa6-414c-8168-07b4de1ad5ae
+      Requested: Int32 IntValue
+        Requested: AutoFixture.Kernel.SeededRequest
+          Requested: System.Int32
+          Created: 195
+        Created: 195
+      Created: 195
+    Created: SampleValueObject
+  Created: SampleValueObject
+```
+
+Behaviors implement the `ISpecimenBuilderTransformation` interface. This is what the interface looks like
+
+```csharp
+public interface ISpecimenBuilderTransformation
+{
+    ISpecimenBuilderNode Transform(ISpecimenBuilder builder);
+}
+```
+
+## Customizations
+
+Finally, AutoFixture uses classes implementing the `ICustomization` interface to wrap in a single place multiple changes to the fixture configuration.
+
+The `ICustomization` interface exposes a single method, `Customize`, that accepts a fixture and applies configuration changes to it.
+
+```csharp
+public interface ICustomization
+{
+    void Customize(IFixture fixture);
+}
+```
+
+For example, the customization below injects values for `int` and `string`.
+
+```csharp
+public class TestCustomization : ICustomization
+{
+    public void Customize(IFixture fixture)
+    {
+        fixture.Inject("Hello world");
+        
+        fixture.Inject(42);
+    }
+}
```

Comment:
How about we add an example customizing an instance of `SampleValueObject` used previously, since I think it's most likely that someone will use this to try and customize a class.

Reply:
Done!